### PR TITLE
Enhance `test_lexer.py` with Tests for New Bitwise Operators and Data Types

### PR DESCRIPTION
**Pull request by Raghav-2903**
### Description:

This PR enhances the `test_lexer.py` file by adding new test cases to cover the recently added support for bitwise operators and additional data types in the lexer. The changes ensure that the lexer correctly identifies and tokenizes the following elements:

**Added Test Cases:**
- **Bitwise Operators:**
  - `&` (Bitwise AND)
  - `|` (Bitwise OR)
  - `^` (Bitwise XOR)
  - `~` (Bitwise NOT)
  - `<<` (Bitwise Shift Left)
  - `>>` (Bitwise Shift Right)

- **Assignment Operations for Bitwise Operators:**
  - `&=` (Assignment AND)
  - `|=` (Assignment OR)
  - `^=` (Assignment XOR)
  - `%=` (Assignment Modulus)
  - `<<=` (Assignment Shift Left)
  - `>>=` (Assignment Shift Right)

- **New Data Types:**
  - `double` (Double Data Type)
  - `unsigned int` (Unsigned Integer Data Type)
  - `const` (Type Qualifier)

**Purpose of the Changes:**
These updates ensure that the lexer correctly processes and identifies tokens related to bitwise operations and additional data types. This is critical for expanding the language's capabilities and ensuring robust parsing of shader code.

**Testing:**
The new test cases have been added to validate the lexer’s functionality with these operators and data types. The tests were executed, and all passed, confirming that the lexer is functioning as expected after the changes.

### CODE REVIEW

**Comment by Raghav-2903** [comment]
@samthakur587 Have made the necessary changes to both the lexer.py and test_lexer.py files

**Comment by samthakur587** [comment]
hii @Raghav-2903  can you mention on which issue are you working on. make the only changes which are mention in your task other task someone else is working on.

File: crosstl/src/translator/lexer.py [hunk 1]
```diff
@@ -64,8 +64,27 @@
...
+    ("ASSIGN_XOR", r"\^="),
+    ("ASSIGN_MODULUS", r"%="),
+    ("ASSIGN_SHIFT_LEFT", r"<<="),
+    ("ASSIGN_SHIFT_RIGHT", r">>="),
+    ("MODULUS", r"%"),
+    ("LOGICAL_AND", r"&&"),
+    ("LOGICAL_OR", r"\|\|"),
+    ("DOUBLE", r"\bdouble\b"),
+    ("UNSIGNED_INT", r"\bunsigned\s+int\b"),
+    ("CONST", r"\bconst\b"),
```
Comment (samthakur587) [inline crosstl/src/translator/lexer.py:84]: the datatype and const is not part of your task i guess.

Comment (Raghav-2903) [reply crosstl/src/translator/lexer.py:84]: Hi @samthakur587 yes I was assigned issues 42-47, I see the mistake made and I will close this PR

**Pull request closed without merging**